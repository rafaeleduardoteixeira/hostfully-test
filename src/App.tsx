import React from 'react';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import createdRoutes from './constants/router';
import { Header } from './components/Header/Header';
import { AppContainer } from './App.styles';

function App() {
  return (
    <AppContainer>
      <Header />
      <RouterProvider router={createdRoutes} />
    </AppContainer>
  );
}

export default App;
