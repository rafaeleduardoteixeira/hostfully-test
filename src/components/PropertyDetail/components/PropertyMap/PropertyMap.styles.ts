import styled from 'styled-components';
import { colors, media } from '../../../../themes';

export const PropertyMapContainer = styled.div`
  width: 70%;
  height: 400px;
  background-color: ${colors.black};
  position: relative;

  .MapStyles {
    height: 400px;
    z-index: 1;
  }

  @media ${media.mobile} {
    width: 100%;
  }
`;
