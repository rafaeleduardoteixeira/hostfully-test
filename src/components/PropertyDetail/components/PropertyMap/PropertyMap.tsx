// Only to emulate the map to compose the layout
import L from 'leaflet';
import { MapContainer, Marker, Popup, TileLayer } from 'react-leaflet';
import 'leaflet/dist/leaflet.css';
import { PropertyMapContainer } from './PropertyMap.styles';
import { Property } from '../../../../interfaces/property';

interface PropertyMapProps {
  property: Property;
}

export const PropertyMap = ({ property }: PropertyMapProps) => {
  const myIcon = new L.Icon({
    iconUrl: '/images/marker.png',
    iconRetinaUrl: '/images/marker.png',
    popupAnchor: [-0, -0],
    iconSize: [25, 40],
  });

  const position: [number, number] = [property?.position.latitude || 0, property?.position.longitude || 0];

  return (
    <PropertyMapContainer>
      <MapContainer center={position} zoom={13} scrollWheelZoom={false} className="MapStyles">
        <TileLayer
          attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />
        <Marker position={position} icon={myIcon}>
          <Popup>{property.name}</Popup>
        </Marker>
      </MapContainer>
    </PropertyMapContainer>
  );
};
