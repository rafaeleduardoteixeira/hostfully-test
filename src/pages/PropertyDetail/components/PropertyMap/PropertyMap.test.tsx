import React, { act } from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { PropertyMap } from './PropertyMap';
import { AppContext } from '../../../../Context/App.context';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock('react-leaflet', () => jest.fn());

describe('PropertyMap', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(
      <AppContext.Provider
        value={{
          isLoading: false,
          getProperties: jest.fn(),
          getPropertyById: jest.fn(),
          getPropertyRentedDates: jest.fn(),
          getRentedProperties: jest.fn(),
          handleAddRentals: jest.fn(),
          handleRemoveRentals: jest.fn(),
          handleEditRentals: jest.fn(),
        }}
      >
        <PropertyMap
          property={{
            id: '1',
            name: 'Name 1',
            description: 'Description 1',
            city: 'City 1',
            country: 'Country 1',
            purpose: 'Purpose 1',
            price: 1,
            images: ['image'],
            position: {
              latitude: 0,
              longitude: 0,
            },
          }}
        />
      </AppContext.Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Should render PropertyMap', () => {
    render(
      <AppContext.Provider
        value={{
          isLoading: false,
          getProperties: jest.fn(),
          getPropertyById: jest.fn(),
          getPropertyRentedDates: jest.fn(),
          getRentedProperties: jest.fn(),
          handleAddRentals: jest.fn(),
          handleRemoveRentals: jest.fn(),
          handleEditRentals: jest.fn(),
        }}
      >
        <PropertyMap
          property={{
            id: '1',
            name: 'Name 1',
            description: 'Description 1',
            city: 'City 1',
            country: 'Country 1',
            purpose: 'Purpose 1',
            price: 1,
            images: ['image'],
            position: {
              latitude: 0,
              longitude: 0,
            },
          }}
        />
      </AppContext.Provider>,
    );

    const PropertyMapElement = screen.getByTestId('PropertyMap');
    expect(PropertyMapElement).toBeInTheDocument();
  });
});
