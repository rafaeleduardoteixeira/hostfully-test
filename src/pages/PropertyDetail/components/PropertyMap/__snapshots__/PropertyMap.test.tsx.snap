// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyMap Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu lnBfiE"
    data-testid="PropertyMap"
  />
</DocumentFragment>
`;
